### PR TITLE
Stdlib build: Only delete out folder contents, not the folder itself

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

Motivation: https://github.com/0xPolygonMiden/miden-vm/issues/1208

I'm not sure if this will fix the issue, but I guess it is worth giving it a try.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: stdlib/build.rs [hunk 1]
```diff
@@ -57,12 +57,24 @@ trait Renderer {
 }
 
 /// Writes Miden standard library modules documentation markdown files based on the available modules and comments.
-pub fn build_stdlib_docs(module_map: &ModuleMap, output_dir: &str) {
-    // Remove functions folder to re-generate
-    fs::remove_dir_all(output_dir).unwrap();
-    fs::create_dir(output_dir).unwrap();
+pub fn build_stdlib_docs(module_map: &ModuleMap, output_dir: &str) -> io::Result<()> {
```
Comment (bobbinth) [inline stdlib/build.rs:60]: Does this function actually get run when running `cargo doc`? I thought that line 24 above should prevent this - but maybe it doesn't work this way?

Comment (hackaugusto) [reply stdlib/build.rs:60]: Here is where I looked:

https://docs.rs/crate/miden-stdlib/latest

image

Comment (hackaugusto) [reply stdlib/build.rs:60]: This seems to be the documented approach https://docs.rs/about/builds#detecting-docsrs

**Comment by hackaugusto** [comment]
can we merge this? is there any reason to not do it?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**